### PR TITLE
Make line separator customisable

### PR DESCRIPTION
**Pull request by LeonB**

### CODE REVIEW

File: decode.go [hunk 1]
```diff
@@ -178,19 +182,44 @@ func findFirstMultiByteChar(data string) int {
...
 
-func (d *Decoder) readLine(v reflect.Value) (err error, ok bool) {
-	line, err := d.data.ReadString('\n')
-	if err != nil && err != io.EOF {
-		return err, false
+func (d *Decoder) SetSeparator(separator string) {
+	d.separator = separator
+}
+
+func (d Decoder) Separator() string {
```
Comment (ianlopshire) [inline decode.go]: I don't see any reason this should be exported. Also is there any reason it is not a pointer receiver e.g. `func (d *Decoder)`?

File: decode.go [hunk 2]
```diff
@@ -178,19 +182,44 @@ func findFirstMultiByteChar(data string) int {
...
-		d.done = true
 
-		if len(line) <= 0 || line[0] == '\n' {
-			// skip last empty lines
-			return nil, false
-		}
+	return "\n"
+}
+
+func (d *Decoder) Scan(data []byte, atEOF bool) (advance int, token []byte, err error) {
```
Comment (ianlopshire) [inline decode.go]: Why implement the custom scan function? I think `data.ReadString(d.lineTerminator)` would work fine.

Comment (LeonB) [reply decode.go]: @ianlopshire `ReadString` only takes a single byte as argument: https://golang.org/pkg/bufio/#Reader.ReadString

So in the case of "\r\n" this wouldn't work

**Comment by LeonB** [comment]
@ianlopshire I've added your suggestions. The suggestion about using `ReadString` instead of `Scan` wouldn't work I think because ReadString only takes a single bytes as an argument: https://golang.org/pkg/bufio/#Reader.ReadString

So in the case of `\r\n` that wouldn't work.

I'm happy to edit the implementation but this was the best I could come up with besides a custom implementation of a `ReadString` like function.

**Comment by ianlopshire** [comment]
@LeonB, thanks for making the changes! merged!

**Pull request merged**